docs/Design/For Each.md

```diff
@@ -0,0 +1,157 @@
+# For Each
+
+A loop that iterates over all elements of an array. 
+This loop does not use an explicit index.
+
+## Commands
+
+### `for each start`
+
+`for each start : array type iterator`
+
+Starts a for each block. 
+The first parameter is the array to be iterated over.
+The second command is the type of the array. 
+The third parameter is the iterator variable.
+
+### `for each end`
+
+Ends a for each block.
+
+## Usage
+
+```
+for each start : myArray int item
```

Comment:
I would consider also changing to `names`/`name` instead of `myArray`/`item`. A little more understandable.